client/components/domains/trademark-claims-notice/index.jsx

```diff
@@ -0,0 +1,563 @@
+					<span className="trademark-claims-notice__claim-item-data">{ courtCase.courtName }</span>
+				</li>
+			</ul>
+		) );
+	};
+
+	renderTrademarkClaimUdrpCases = udrpCases => {
+		if ( ! udrpCases ) {
+			return;
+		}
+
+		return udrpCases.map( ( udrpCase, index ) => (
+			<ul
+				className="trademark-claims-notice__claim-item-list trademark-claims-notice__claim-item"
+				key={ index }
+			>
+				<li key="case-no">
+					<span className="trademark-claims-notice__claim-item-label">Case Number: </span>
+					<span className="trademark-claims-notice__claim-item-data">{ udrpCase.caseNo }</span>
+				</li>
+				<li key="udrp-provider">
+					<span className="trademark-claims-notice__claim-item-label">UDRP Provider: </span>
+					<span className="trademark-claims-notice__claim-item-data">
+						{ udrpCase.udrpProvider }
+					</span>
+				</li>
+			</ul>
+		) );
+	};
+
+	renderTrademarkClaimCases = claim => {
+		const { notExactMatch } = claim;
+
+		if ( ! notExactMatch ) {
+			return;
+		}
+
+		let { court: courtCases, udrp: udrpCases } = notExactMatch;
+		if ( courtCases && ! Array.isArray( courtCases ) ) {
+			courtCases = [ courtCases ];
+		}
+		if ( udrpCases && ! Array.isArray( udrpCases ) ) {
+			udrpCases = [ udrpCases ];
+		}
+
+		// Note: Please don't update the language rendered here as it is specified in the RPM-Requirements of the
+		// ICANN TMCH functional specifications in: https://tools.ietf.org/html/draft-ietf-eppext-tmch-func-spec-01#ref-RPM-Requirements
+		return (
+			<div className="trademark-claims-notice__claim-item" key="claim-cases">
+				<p>
+					This domain name label has previously been found to be used or registered abusively
+					against the following trademarks according to the referenced decisions:
+				</p>
+				{ courtCases && this.renderTrademarkClaimCourtCases( courtCases ) }
+				{ udrpCases && this.renderTrademarkClaimUdrpCases( udrpCases ) }
+			</div>
+		);
+	};
+
+	renderTrademarkClaimsInfo = () => {
+		const { trademarkClaimsNoticeInfo } = this.props;
+		let { claim: claims } = trademarkClaimsNoticeInfo;
+
+		if ( ! Array.isArray( claims ) ) {
+			claims = [ claims ];
+		}
+
+		return (
+			<ol>
+				{ claims.map( ( claim, index ) => (
+					<li className="trademark-claims-notice__claim" key={ index }>
+						{ this.renderTrademarkClaimMark( claim ) }
+						{ this.renderTrademarkClaimJurisdiction( claim ) }
+						{ this.renderTrademarkClaimGoodsAndServices( claim ) }
+						{ this.renderTrademarkClaimInternationalClassification( claim ) }
+						{ this.renderTrademarkClaimRegistrant( claim ) }
+						{ this.renderTrademarkClaimContact( claim ) }
+						{ this.renderTrademarkClaimCases( claim ) }
+					</li>
+				) ) }
+			</ol>
+		);
+	};
+
+	// Note: Please don't update the language rendered here as it is specified in the RPM-Requirements of the
+	// ICANN TMCH functional specifications in: https://tools.ietf.org/html/draft-ietf-eppext-tmch-func-spec-01#ref-RPM-Requirements
+	renderNotice() {
```

Comment:
I wonder if starting from this, this should not be a separate component that just accepts the claims info and some callbacks for the actions - the component file is long as-is and this legalese seems like something that would be good to put in a separate component, since it has nothing to do with the rest of the state as far as I can see.